### PR TITLE
chore: bump ethers

### PR DESCRIPTION
**Pull request by gakonst**
As title! You may also wanna chase @ralexstokes to update ENR in the consensus crate. I think ENR can't be bumped yet because of some libp2p issue.

```
error: failed to select a version for `curve25519-dalek`.
    ... required by package `snow v0.9.1`
    ... which satisfies dependency `snow = "^0.9.0"` (locked to 0.9.1) of package `libp2p-noise v0.41.0`
    ... which satisfies dependency `libp2p-noise = "^0.41.0"` (locked to 0.41.0) of package `libp2p v0.50.0`
    ... which satisfies dependency `libp2p = "^0.50.0"` (locked to 0.50.0) of package `lighthouse_network v0.2.0 (/Users/georgios/paradigm/lighthouse/beacon_node/lighthouse_network)`
    ... which satisfies path dependency `lighthouse_network` (locked to 0.2.0) of package `beacon_node v3.5.1 (/Users/georgios/paradigm/lighthouse/beacon_node)`
    ... which satisfies path dependency `beacon_node` (locked to 3.5.1) of package `boot_node v3.5.1 (/Users/georgios/paradigm/lighthouse/boot_node)`
    ... which satisfies path dependency `boot_node` (locked to 3.5.1) of package `lighthouse v3.5.1 (/Users/georgios/paradigm/lighthouse/lighthouse)`
versions that meet the requirements `=4.0.0-rc.0` (locked to 4.0.0-rc.0) are: 4.0.0-rc.0

all possible versions conflict with previously selected packages.

  previously selected package `curve25519-dalek v4.0.0-rc.1`
    ... which satisfies dependency `curve25519-dalek = "=4.0.0-rc.1"` of package `ed25519-dalek v2.0.0-pre.0`
    ... which satisfies dependency `ed25519-dalek = "^2.0.0-pre.0"` of package `enr v0.8.0`
    ... which satisfies dependency `enr = "^0.8.0"` (locked to 0.8.0) of package `eth2_network_config v0.2.0 (/Users/georgios/paradigm/lighthouse/common/eth2_network_config)`
    ... which satisfies path dependency `eth2_network_config` (locked to 0.2.0) of package `account_manager v0.3.5 (/Users/georgios/paradigm/lighthouse/account_manager)`
    ... which satisfies path dependency `account_manager` (locked to 0.3.5) of package `lighthouse v3.5.1 (/Users/georgios/paradigm/lighthouse/lighthouse)`

failed to select a version for `curve25519-dalek` which could resolve this conflict
```

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1988,7 +2047,25 @@ dependencies = [
  "bytes",
  "ed25519-dalek",
  "hex",
- "k256",
+ "k256 0.11.6",
```
Comment (gakonst) [inline Cargo.lock]: You guys may also want to bump K256 & related crates, here's the ethers PR with some indications on the breakages https://github.com/gakonst/ethers-rs/pull/2260

**Review by gakonst — COMMENTED** [review]
I guess there's some dep bumping to do in the discv5 as well

**Comment by paulhauner** [comment]
Sorry to leave you hanging here @gakonst. I've changed the target branch to `unstable` (unfortunately we can't have one branch on the homepage and another as the default PR target) and pushed a commit to force Github to re-check the target branch.

I also noticed a conflict, so I took the liberty to fix it. I hope you don't mind.

Let's see how CI likes it.

**Comment by paulhauner** [comment]
Looks like we have some compile issues here. Keen to get it resolved eventually, but I'll bump this from v4.2.0 since we're keen to get it out and this isn't urgent AFAICT.

**Comment by paulhauner** [comment]
I've spent another couple of hours trying to figure out why these tests are failing, but I'm at a loss. I'm going to put it down for now, but I'll leave some notes for next time.

This is the test failure (it only fails on Linux since this endpoint is disabled on other platforms):

```bash
---- tests::lighthouse_endpoints stdout ----
thread 'tests::lighthouse_endpoints' panicked at 'called `Result::unwrap()` on an `Err` value: HttpClient(, kind: decode, detail: invalid type: map, expected f32 at line 1 column 920)', beacon_node/http_api/tests/tests.rs:3999:51
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
````

It's failing to decode the output of a `/lighthouse/health` query. Here's an example of some failing output:

```json
{"data":{"sys_virt_mem_total":134823710720,"sys_virt_mem_available":114806923264,"sys_virt_mem_used":18717569024,"sys_virt_mem_free":35534340096,"sys_virt_mem_percent":14.846638,"sys_virt_mem_cached":76801110016,"sys_virt_mem_buffers":3770691584,"sys_loadavg_1":5.71,"sys_loadavg_5":3.66,"sys_loadavg_15":6.94,"cpu_cores":24,"cpu_threads":32,"system_seconds_total":14176,"user_seconds_total":265113,"iowait_seconds_total":30194,"idle_seconds_total":583831,"cpu_time_total":865718,"disk_node_bytes_total":977536479232,"disk_node_bytes_free":787077062656,"disk_node_reads_total":98319033,"disk_node_writes_total":74747481,"network_node_bytes_total_received":110675959252,"network_node_bytes_total_transmit":54991646254,"misc_node_boot_ts_seconds":1692569914,"misc_os":"linux","pid":3138448,"pid_num_threads":69,"pid_mem_resident_set_size":98078720,"pid_mem_virtual_memory_size":4752084992,"pid_process_seconds_total":5}}
```

I've compared that JSON to an example from the `stable` branch and I don't see a functional difference:

```json
{"data":{"sys_virt_mem_total":134823710720,"sys_virt_mem_available":114300575744,"sys_virt_mem_used":19223687168,"sys_virt_mem_free":30878449664,"sys_virt_mem_percent":15.2222,"sys_virt_mem_cached":80905175040,"sys_virt_mem_buffers":3816398848,"sys_loadavg_1":3.53,"sys_loadavg_5":5.03,"sys_loadavg_15":5.88,"cpu_cores":24,"cpu_threads":32,"system_seconds_total":14356,"user_seconds_total":267338,"iowait_seconds_total":30310,"idle_seconds_total":601829,"cpu_time_total":886146,"disk_node_bytes_total":977536479232,"disk_node_bytes_free":783827701760,"disk_node_reads_total":98486994,"disk_node_writes_total":75573767,"network_node_bytes_total_received":114172478540,"network_node_bytes_total_transmit":55513922654,"misc_node_boot_ts_seconds":1692569914,"misc_os":"linux","pid":3177675,"pid_num_threads":69,"pid_mem_resident_set_size":104931328,"pid_mem_virtual_memory_size":4747530240,"pid_process_seconds_total":5}}
```

Therefore, I suspect that something in the decoding of that JSON has changed in this branch. I haven't figured out what that is yet 🤔

**Comment by dapplion** [comment]
@gakonst should we switch to new alloy lib now?

**Comment by michaelsproul** [comment]
closing in favour of alloy:

- https://github.com/sigp/lighthouse/pull/5301

**Pull request closed without merging**